### PR TITLE
Release version 1.21

### PR DESCRIPTION
**Pull request by Stigmi**
Release tested & validated by bpost.
Change-log here: http://www.stigmi.eu/bpost-module-prestashop/change-log.html

### CODE REVIEW

File: bpostshm.php [hunk 1]
```diff
@@ -47,13 +48,27 @@ public function __construct()
...
+			'beforeCarrier',			// displayBeforeCarrier
+			'extraCarrier',				// displayCarrierList
+			'paymentTop',				// displayPaymentTop
+			'processCarrier',			// actionCarrierProcess
+			'newOrder',					// actionValidateOrder
+			'postUpdateOrderStatus',	// actionOrderStatusPostUpdate
+			'updateCarrier',			// actionCarrierUpdate
+			// v1.21
+			// 'adminOrder',				// displayAdminOrder
+			'orderDetailDisplayed',		// displayOrderDetail
```
Comment (Quetzacoalt91) [inline bpostshm.php]: You add a new hook with this new version. You should implement an upgrade file to register it when the module is already installed

File: bpostshm.php [hunk 2]
```diff
@@ -1191,6 +1260,58 @@ public function hookPostUpdateOrderStatus($params)
...
+				}
+
+				$this->context->smarty->assign('module_dir', __PS_BASE_URI__.'/modules/'.$this->name);
+				$this->context->smarty->assign('bpost', $bpost);
+
+				return $this->context->smarty->fetch(dirname(__FILE__).'/views/templates/admin/admin_order_details.tpl');
+			}
+		}
+
+		return 'not ours';
```
Comment (Quetzacoalt91) [inline bpostshm.php]: Do you really need to return something of the order is not yours ?

File: views/templates/admin/admin_order_details.tpl [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{*
+* 2014-2015 Stigmi
+*
+* @author Serge <serge@stigmi.eu>
+* @copyright 2014-2015 Stigmi
+* @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+*}
+
+{if isset($sp)}
+<div style="background: url({$module_dir|escape}/views/img/sp.png) no-repeat;background-size:auto 70px;padding:20px 0px;">
```
Comment (Quetzacoalt91) [inline views/templates/admin/admin_order_details.tpl]: Please do not use style inline. Moive you style definition in a CSS file.

File: bpostshm.php (hunk 1 above)
Comment (Stigmi) [inline bpostshm.php]: OK done

File: views/templates/admin/admin_order_details.tpl (hunk 3 above)
Comment (Stigmi) [inline views/templates/admin/admin_order_details.tpl]: OK done

File: bpostshm.php (hunk 2 above)
Comment (Stigmi) [inline bpostshm.php]: OK, modified

File: upgrade/install-1.21.0.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	$new_hooks = array(
+			'orderDetailDisplayed',		// displayOrderDetail
+		);
+
+	/*  Try to register the new hook since last version */
+	foreach ($new_hooks as $hook)
+			if (!$object->isRegisteredInHook($hook))
+				$object->registerHook($hook);
+
+	return (bool)count($object->upgrade_detail[$upgrade_version]);
```
Comment (Quetzacoalt91) [inline upgrade/install-1.21.0.php]: If I understand the code you wrote here, it seems that this line will return 0.

If the upgrade went well, you must return ```true``` or the module will be disabled.

Comment (Stigmi) [inline upgrade/install-1.21.0.php]: Indeed there was a problem, the hook was registered but there was an error message + the module was disabled. Thank you for noticing it, we updated the file and changed the way we tested the upgrade file.

**Comment by Quetzacoalt91** [comment]
Hi @Stigmi,

Thanks, your PR has just been merged and the changes should be available from tomorrow.

Best regards

**Pull request merged**